src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -233,15 +254,24 @@ fun collectPathResolveVariants(
     f: (RsResolveProcessor) -> Unit
 ): List<RsPathResolveResult<RsElement>> {
     val referenceName = path.referenceName ?: return emptyList()
-    val result = SmartList<RsPathResolveResult<RsElement>>()
-    val processor = createProcessor(referenceName) { e ->
-        if (e.name == referenceName) {
-            collectPathScopeEntry(ctx, result, e)
+    val processor = SinglePathResolveVariantsCollector(ctx, referenceName)
+    f(processor)
+    return processor.result
+}
+
+private class SinglePathResolveVariantsCollector(
```

Comment:
Do I understand correctly that this class is extracted to get user-readable names in stack traces?